src/test/pybind/test_rbd.py

```diff
@@ -379,40 +391,39 @@ def test_remove_dne():
 def test_list_empty():
     eq([], RBD().list(ioctx))
 
-@with_setup(create_image, remove_image)
-def test_list():
-    eq([image_name], RBD().list(ioctx))
+def test_list(tmp_image):
+    name = tmp_image
+    eq([name], RBD().list(ioctx))
```

Comment:
This touches the test body, unnecessarily IMO.  Ditching the global would be nice, but since it's not done in full in this PR, it's better to stick to it for consistency.

Reply:
Scrolling down, the same goes for a few other tests.  I would suggest the following incremental for `src/test/pybind/test_rbd.py`:
```
diff --git a/src/test/pybind/test_rbd.py b/src/test/pybind/test_rbd.py
index 507c48f32d12..d67b00e480d2 100644
--- a/src/test/pybind/test_rbd.py
+++ b/src/test/pybind/test_rbd.py
@@ -121,9 +121,9 @@ def remove_image():
 
 @pytest.fixture
 def tmp_image():
-    name = create_image()
-    yield name
-    RBD().remove(ioctx, name)
+    create_image()
+    yield
+    remove_image()
 
 def create_group():
     global group_name
@@ -137,9 +137,9 @@ def remove_group():
 
 @pytest.fixture
 def tmp_group():
-    name = create_group()
-    yield name
-    RBD().group_remove(ioctx, name)
+    create_group()
+    yield
+    remove_group()
 
 def rename_group():
     new_group_name = "new" + group_name
@@ -392,38 +392,36 @@ def test_list_empty():
     eq([], RBD().list(ioctx))
 
 def test_list(tmp_image):
-    name = tmp_image
-    eq([name], RBD().list(ioctx))
+    eq([image_name], RBD().list(ioctx))
 
-    with Image(ioctx, name) as image:
+    with Image(ioctx, image_name) as image:
         image_id = image.id()
     eq([{'id': image_id, 'name': image_name}], list(RBD().list2(ioctx)))
 
 def test_remove_with_progress():
-    name = create_image()
+    create_image()
     d = {'received_callback': False}
     def progress_cb(current, total):
         d['received_callback'] = True
         return 0
 
-    RBD().remove(ioctx, name, on_progress=progress_cb)
+    RBD().remove(ioctx, image_name, on_progress=progress_cb)
     eq(True, d['received_callback'])
 
 def test_remove_canceled(tmp_image):
     def progress_cb(current, total):
         return -ECANCELED
 
-    assert_raises(OperationCanceled, RBD().remove, ioctx, tmp_image,
+    assert_raises(OperationCanceled, RBD().remove, ioctx, image_name,
                   on_progress=progress_cb)
 
 def test_rename(tmp_image):
     rbd = RBD()
-    image_name1 = tmp_image
     image_name2 = get_temp_image_name()
-    rbd.rename(ioctx, image_name1, image_name2)
+    rbd.rename(ioctx, image_name, image_name2)
     eq([image_name2], rbd.list(ioctx))
-    rbd.rename(ioctx, image_name2, image_name1)
-    eq([image_name1], rbd.list(ioctx))
+    rbd.rename(ioctx, image_name2, image_name)
+    eq([image_name], rbd.list(ioctx))
 
 def test_pool_metadata():
     rbd = RBD()
@@ -2532,7 +2530,7 @@ def test_list_groups_empty():
     eq([], RBD().group_list(ioctx))
 
 def test_list_groups(tmp_group):
-    eq([tmp_group], RBD().group_list(ioctx))
+    eq([group_name], RBD().group_list(ioctx))
 
 def test_list_groups_after_removed():
     create_group()
```